File: Server/Server/ecommerce_backend/Models/order.cs
using System.ComponentModel.DataAnnotations;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace WebServerWithMongoDB.Models
{
public class Order
{
    public ObjectId Id { get; set; }
    public string? OrderId { get; set; } 
    public string? CustomerId { get; set; }

    [Required]
    public List<Product>? Products { get; set; }
    public string? Status { get; set; } 
    public DateTime CreatedAt { get; set; }
}
}

File: MobileApp/ECommerceApp/app/src/main/java/com/example/ecommerceapp/models/Order.kt
package com.example.ecommerceapp.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Order(
    val OrderId: String,                 // Order ID
    val customerId: String,              // ID of the customer who placed the order
    val products: List<Product>,          // Change this to a list of Product objects
    val isPurchased: Boolean,             // Whether the order has been purchased
    val status: String,                   // Order status (e.g., Pending, Shipped, Delivered)
    val createdAt: String                 // Date of order creation as a string (could be a Date type)
) : Parcelable
